services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/HankeControllerITests.kt

```diff
@@ -129,4 +132,42 @@ class HankeControllerITests(@Autowired val mockMvc: MockMvc) {
 
     }
 
+    @Test
+    fun `Add Hanke and HankeYhteystiedot and return it with newly created hankeTunnus (POST)`() {
+        val hankeName = "Mannerheimintien remontti remonttinen"
+
+        var hankeToBeMocked = Hanke(id = null, hankeTunnus = null, nimi = hankeName, kuvaus = "lorem ipsum dolor sit amet...",
+                onYKTHanke = false, alkuPvm = getCurrentTimeUTC(), loppuPvm = getCurrentTimeUTC(), vaihe = Vaihe.OHJELMOINTI,
+                version = null, createdBy = "Tiina", createdAt = null, modifiedBy = null, modifiedAt = null, saveType = SaveType.DRAFT)
+
+        //HankeYhteystieto Omistaja added
+        hankeToBeMocked.omistajat = arrayListOf(
+                HankeYhteystieto(null, ContactType.TOTEUTTAJA, "Pekkanen", "Pekka",
+                        "pekka@pekka.fi", "3212312", null,
+                        "Kaivuri ja mies", null, null, null,
+                        null, null))
+
+        //changing some return values
+        val expectedHanke = hankeToBeMocked
+                .apply { hankeTunnus = mockedHankeTunnus }
```

Comment:
"apply" does not need to be chained like this - just one is enough and all these separate clauses can be inside it

Reply:
Thanks.

Reply:
Fixed like that